src/chat.ts

```diff
@@ -174,186 +227,133 @@ export class ChatProvider implements vscode.WebviewViewProvider {
-        this.addUserResult(result);
+        this._view.webview.postMessage({
+          command: "chat/addResult",
+          result,
+        });
       } else {
-        this.addAssistantResult(result);
+        if (result.outcome === ChatResultOutcome.Finished) {
+          this._view.webview.postMessage({ command: "chat/assistantFinished" });
+        }
+
+        this._view.webview.postMessage({
+          command: "chat/addResult",
+          result,
+        });
       }
     } else {
       this._unhandledMessages.push(result);
     }
   }
 
-  private addUserResult(result: ChatResult) {
-    this._view.webview.postMessage({
-      command: "add_result",
-      result: {
-        role: result.role,
-        outcome: result.outcome,
-        textContent: result.textContent,
-      },
-    });
-  }
-
-  private addAssistantResult(result: ChatResult) {
-    if (result.outcome === ChatResultOutcome.Finished) {
-      this._currentAssistantMessage = "";
-      this._view.webview.postMessage({ command: "assistant_finished" });
-      return;
-    }
-
-    if (result.outcome === ChatResultOutcome.Success) {
-      this._currentAssistantMessage += result.textContent;
-    } else {
-      this._currentAssistantMessage = result.textContent;
+  public addReviewResult(result: ChatResult) {
+    switch (result.outcome) {
+      case ChatResultOutcome.Finished:
+        this._view.webview.postMessage({ command: "review/assistantFinished" });
+        break;
+      default:
+        this._view.webview.postMessage({
+          command: "review/addResult",
+          result: result,
+        });
     }
+  }
 
-    let sanitized = renderMarkdownMessage(this._currentAssistantMessage);
-
+  public addRecipes(result: Recipe[]) {
+    this.recipes = result;
     this._view.webview.postMessage({
-      command: "add_result",
-      result: {
-        role: result.role,
-        outcome: result.outcome,
-        textContent: sanitized,
-      },
+      command: "recipes/addRecipes",
+      result: result,
     });
   }
 
+  public populateBranches(branches: GitBranches) {
+    console.log("branches populated: ", branches);
+    this.branches = branches;
+  }
+
   public clearChat() {
-    this._view.webview.postMessage({ command: "clear_chat" });
-    this._currentAssistantMessage = "";
+    this._view.webview.postMessage({ command: "chat/clear" });
+  }
+
+  public clearReview() {
+    this._view.webview.postMessage({ command: "review/clear" });
   }
 
   private async _getHtmlForWebview(webview: vscode.Webview) {
     // Get the local path to main script run in the webview, then convert it to a uri we can use in the webview.
-    const scriptUri = webview.asWebviewUri(
-      vscode.Uri.joinPath(this._extensionUri, "src", "webview", "chat.js")
-    );
-
-    // Do the same for the stylesheet.
-    const styleResetUri = webview.asWebviewUri(
-      vscode.Uri.joinPath(this._extensionUri, "media", "reset.css")
+    const baseSrc = webview.asWebviewUri(
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏Rename some of these variables

It's not clear what is what with some of these. Are they ide style/interface implementations or the main coding assistant webapp:
- baseSrc
- scriptSrc
- bridgeSrc
- styleSheetSrc

Reply:
I've renamed some of these and added comments to help clarify.